sync/rpc/server.go

```diff
@@ -26,6 +31,12 @@ type SyncAgentServer struct {
 	endPort         int
 	processesByPort map[int]string
 	backupList      sync.Map
+	backupIDs       BackupIDList
+}
+
+type BackupIDList struct {
+	completedBackups ring.Ring
+	onGoingBackups   ring.Ring
```

Comment:
onGoingBackups shouldn't be a ring buffer.

Reply:
The backupStatus (backupList + backupIDs) should have a defined CRUD interface to keep at most 5 completed backups + unlimited ongoing backup status.

```
Create()
Get()
Update()
Delete()
```

And this struct needs to be unit tested separately to make sure it's implemented correctly. You should able to implement it using a slice (array). No need for `golang-ring`.

The slice has the ordered elements. In fact, since the array should be small, we don't even need a map. Just iterate the array to find the object. You would need an RWMutex for the array.